### PR TITLE
Test translation of view with namesake column aliases which are differing in case

### PR DESCRIPTION
**Pull request by findinpath**
## Description

In the translation of Hive views with Coral,
when joining tables that have the namesake column names,
irrespective of their case, make sure that the resulting
relation is using names of the columns which are 
unique in order to avoid the situation where the
SQL statement created contains ambiguous column names.

> Is this change a fix, improvement, new feature, refactoring, or other?

Test

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

This change applies to Hive and specifically to the translation of Hive views.

> How would you describe this change to a non-technical end user or system administrator?

Hive translation of views that joined tables having namesake column names spelled in different cases failed before because of ambiguous column name reference. The upgrade of the Coral library contains a fix for such usecases. 

## Related issues, pull requests, and links

Coral PR: https://github.com/linkedin/coral/pull/234

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# Hive
* Coral view translation uses unique column names for namesake columns ({issue}`11240`)
```

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java [hunk 1]
```diff
@@ -664,6 +664,28 @@ public void testViewReferencingHiveAndIcebergTables()
         onHive().executeQuery("DROP TABLE default.view_iceberg_table");
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testNamesakeColumnsShouldGetUniqueIdentifiersDuringTheTranslationToAvoidAmbiguity()
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS test_hive_namesake_column_name_a");
+        onHive().executeQuery("DROP TABLE IF EXISTS test_hive_namesake_column_name_b");
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java:671]: Add drops at the end

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java [hunk 2]
```diff
@@ -664,6 +664,28 @@ public void testViewReferencingHiveAndIcebergTables()
         onHive().executeQuery("DROP TABLE default.view_iceberg_table");
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testNamesakeColumnsShouldGetUniqueIdentifiersDuringTheTranslationToAvoidAmbiguity()
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java:668]: testViewWithColumnAliasesDifferingInCase ?

**Review by findepi — APPROVED** [review]

**Comment by findinpath** [comment]
Rebased on `master` to make use of the coral-trino library update which fixes the problem showcased by the test case from this PR.

**Comment by findepi** [comment]
Can you please rename commit and PR similar to https://github.com/trinodb/trino/pull/11240#discussion_r816632786 ?

**Pull request merged**